zebra-consensus/src/transaction/check.rs

```diff
@@ -4,18 +4,49 @@
 
 use std::{borrow::Cow, collections::HashSet, convert::TryFrom, hash::Hash};
 
+use chrono::Utc;
+
 use zebra_chain::{
     amount::{Amount, NonNegative},
     block::Height,
     orchard::Flags,
     parameters::{Network, NetworkUpgrade},
     primitives::zcash_note_encryption,
     sapling::{Output, PerSpendAnchor, Spend},
-    transaction::Transaction,
+    transaction::{LockTime, Transaction},
 };
 
 use crate::error::TransactionError;
 
+/// Checks if the transaction's lock time allows this transaction to be included in a block.
+///
+/// Consensus rule:
+///
+/// > The transaction must be finalized: either its locktime must be in the past (or less
+/// > than or equal to the current block height), or all of its sequence numbers must be
+/// > 0xffffffff.
+///
+/// [`Transaction::lock_time`] validates the transparent input sequence numbers, returning [`None`]
+/// if they indicate that the transaction is finalized by them. Otherwise, this function validates
+/// if the lock time is in the past.
+pub fn lock_time_has_passed(tx: &Transaction, height: Height) -> Result<(), TransactionError> {
```

Comment:
Optional Naming: The comparisons are easier to read if the block data is explicitly named.

```suggestion
pub fn lock_time_has_passed(tx: &Transaction, block_height: Height, block_time: _) -> Result<(), TransactionError> {
```

Reply:
Done in [7d77ba4](https://github.com/jvff/zebra/commit/7d77ba428faf3fbdd27f493843d1e0c9b160f4b6).